slimt/Annotation.cc

```diff
@@ -74,4 +79,81 @@ void AnnotatedText::record_existing_sentence(
   annotation.token_begin_.push_back(text.size());
 }
 
+void AnnotatedText::to(Encoding encoding) {
+  if (encoding == encoding_) return;
+  if (encoding == Encoding::Byte) {
+    WordIterator current(*this);
+
+    std::vector<Range> words;
+
+    size_t utf8_idx = 0;
+    size_t byte_idx = 0;
+    Range utf8{.begin = utf8_idx, .end = 0};
+    for (; byte_idx < text.length(); byte_idx++) {
+      if (byte_idx == (*current).end) {
+        utf8.end = utf8_idx;
+        words.push_back(utf8);
+        ++current;
+        utf8.begin = utf8_idx;
+      }
+    }
+    annotation.update(words);
```

Comment:
I think this is broken:

1. The if condition needs to have `source == src && target == tgt` to be future proof and _complete_.
2. We're not tracking the unicode continuation character to selectively adjust the unicode pointer, so the mappings are wrong.

This should surface in tests, IIUC.